ci/setup-and-build.ps1

```diff
@@ -0,0 +1,164 @@
+            pushd "Plugins"
+            Start-Process -Wait -PassThru -NoNewWindow -FilePath "git" -ArgumentList @(`
+                "clone", `
+                "git@github.com:spatialos/UnrealGDK.git", `
+                "--depth 1", `
+                "-b $gdk_branch_name"
+            )
+            popd
+        popd
+    Finish-Event "clone-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+
+    Start-Event "set-up-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+        pushd "Game/Plugins/UnrealGDK"
+            # Set the required variables for the GDK's setup script to use
+            $gdk_home = Convert-Path .
+            $msbuild_exe = "${env:ProgramFiles(x86)}\MSBuild\14.0\bin\MSBuild.exe"
+
+            # Invoke the setup script
+            &"$($game_home)\Game\Plugins\UnrealGDK\ci\setup-gdk.ps1"
+        popd
+    Finish-Event "set-up-gdk-plugin" "build-gdk-third-person-shooter-:windows:"
+
+    # Set LINUX_MULTIARCH_ROOT and then reload it for this script
+    $clang_path = "$($game_home)\UnrealEngine\ClangToolchain\"
+    [Environment]::SetEnvironmentVariable("LINUX_MULTIARCH_ROOT", $clang_path, "Machine")
+    $env:LINUX_MULTIARCH_ROOT = [System.Environment]::GetEnvironmentVariable("LINUX_MULTIARCH_ROOT", "Machine")
+
+    # Allow the GDK plugin to find the engine
+    $env:UNREAL_HOME = "$($game_home)\UnrealEngine\"
+
+    Start-Event "build-editor" "build-gdk-third-person-shooter-:windows:"
+        # Build the project editor to allow the snapshot commandlet to run
+        $build_editor_proc = Start-Process -PassThru -NoNewWindow -FilePath "$($game_home)\Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat" -ArgumentList @(`
+            "ThirdPersonShooterEditor", `
+            "Win64", `
+            "Development", `
+            "ThirdPersonShooter.uproject"
+        )
+        $build_editor_handle = $build_editor_proc.Handle
+        Wait-Process -Id (Get-Process -InputObject $build_editor_proc).id
+        if ($build_editor_proc.ExitCode -ne 0) {
+            Write-Log "Failed to build Win64 Development Editor. Error: $($build_editor_proc.ExitCode)"
+            Throw "Failed to build Win64 Development Editor"
+        }
+    Finish-Event "build-editor" "build-gdk-third-person-shooter-:windows:"
+
+    # Invoke the GDK commandlet to generate schema and snapshot. Note: this needs to be run prior to cooking 
+    Start-Event "generate-schema" "build-gdk-third-person-shooter-:windows:"
+        pushd "UnrealEngine/Engine/Binaries/Win64"
+            Start-Process -Wait -PassThru -NoNewWindow -FilePath ((Convert-Path .) + "\UE4Editor.exe") -ArgumentList @(`
+                "$($game_home)/Game/ThirdPersonShooter.uproject", `
+                "-run=GenerateSchemaAndSnapshots", `
+                "-MapPaths=`"/Maps/TPS-Start_Small`""
+            )
+
+            $core_gdk_schema_path = "$($game_home)\Game\Plugins\UnrealGDK\SpatialGDK\Extras\schema\*"
```

Comment:
This logic is very GDK dependent. Perhaps we should invoke some scripts within the gdk directory directly.